cdc/sink/mq/mq.go

```diff
@@ -347,6 +347,11 @@ func (k *mqSink) EmitDDLEvent(ctx context.Context, ddl *model.DDLEvent) error {
 // table sinks before closing it. So there is no writing after closing.
 func (k *mqSink) Close(ctx context.Context) error {
 	k.resolvedBuffer.Close()
+	// We must finish consuming the data here,
+	// otherwise it will cause the channel to not close properly.
+	for range k.resolvedBuffer.Out() {
```

Comment:
Is there a possible this for loop will block the `processorManager`.

Reply:
We stop sending data before we stop sink. so I don't think so, and it doesn't make sense for the consumer to get stuck after closing?